include/triton/Analysis/Utility.h

```diff
@@ -20,6 +20,18 @@ template <typename Int> Int product(llvm::ArrayRef<Int> arr) {
 
 template <typename Int> Int ceil(Int m, Int n) { return (m + n - 1) / n; }
 
+// output[i] = input[order[i]]
+template <typename T>
+SmallVector<T> reorder(ArrayRef<T> input, ArrayRef<unsigned> order) {
+  size_t rank = order.size();
+  assert(input.size() == rank);
+  SmallVector<T> result(rank);
+  for (auto it : llvm::enumerate(order)) {
+    result[it.index()] = input[it.value()];
```

Comment:
Nit: is it necessary to `assert` that `it.index() < rank`?

Reply:
Yeah, normally the index should be take care of. But the `order` is a core data structure widely used across the whole compile pipeline, I think the upstream stages has guarded well with it, and it should work fine with vectors of the same size.